### ISSUE TITLE
[Button bar] Update the button bar documentation

### ISSUE THREAD

**Issue by anna-lach** [sl-design-system/components#2843]
### Provide a general summary of the issue here

Since there is no `role` added to the `button-bar`, the `aria-label` will not work properly with assistive technologies. 

So the examples and accessibility part in the documentation website need to be updated.

The button bar is a purely visual container, it doesn't have any meaning, so there will be no `role` there.

### 👤 Your name

Anna Sobczak

### 🧢 Your Product/Team

SLDS
